app/cli/main.go

```diff
@@ -16,25 +16,75 @@
 package main
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"os"
+	"strings"
 
 	"github.com/chainloop-dev/chainloop/app/cli/cmd"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/posthog"
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1"
 	"github.com/go-kratos/kratos/v2/errors"
 	jwtMiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
 	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc/status"
 )
 
+// Follow the convention stated on https://consoledonottrack.com/
+const doNotTrackEnv = "DO_NOT_TRACK"
+
+var (
+	// Posthog API key and endpoint to be injected in the build process
+	posthogAPIKey   = ""
+	posthogEndpoint = ""
+)
+
 func main() {
 	// Couldn't find an easier way to disable the timestamp
 	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr, FormatTimestamp: func(interface{}) string { return "" }})
 	rootCmd := cmd.NewRootCmd(logger)
 
-	if err := rootCmd.Execute(); err != nil {
-		msg, exitCode := errorInfo(err, logger)
+	// Set the default exit code to 0, in case of an error it will be updated
+	exitCode := 0
+
+	// Run the command and get the executed command
+	executedCmd, err := rootCmd.ExecuteC()
+	if err != nil {
+		var msg string
+		msg, exitCode = errorInfo(err, logger)
 		logger.Error().Msg(msg)
-		os.Exit(exitCode)
+	}
+
+	runTelemetry(logger, executedCmd, exitCode)
```

Comment:
I see that only succeeded commands will be tracked.
Do you thins this could be moved to the `PreRunE` (or PostRun) method in the rootCmd command instead?

Reply:
No no, it will track both successful and unsuccessful commands that's why it's moved into the main package. The problem with the `PostRun(E)` is that they don't run if there has been an error. Moving the logic to the main caller, `rootCmd.Execute` give us the chance to hook it up to know when there has been an error and track it as well.

Additionally to know what's the actual command that has been run we moved it to use `rootCmd.ExecuteC` which returns the executed command therefore in this way we have the whole picture :)

Reply:
but to be clear, there is a bug in that code that doesn't do that right? That was a catch from @jiparis.

Reply:
Sorry, my bad, I didn't notice the removal of `os.Exit(exitCode)`. Ignore my comment.